app/Http/Controllers/TourController.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Tour;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+
+class TourController extends Controller
+{
+    /**
+     * Display a listing of the resource having possibility to choose specific filter
+     * 
+     * 
+     * @return $data - specific or general data from tours
+     * @param $request - requests from url
+     */
+    public function index(Request $request)
+    {
+        $data = DB::table('tours');
+
+        if(!empty($request->all())){
+
+        if($request->has('price.eq')){
+           
+        $data->where('price', '=', $request->all());
+
+    }elseif($request->has('price.gte')){
+
+        $data->where('price', '>=', $request->all());
+
+    }elseif($request->has('price.lte')){
+
+        $data->where('price', '<=', $request->all());
+
+    }elseif($request->has('start.eq')){
+
+        $data->where('start', '=', $request->all());
+
+    }elseif($request->has('start.gte')){
+
+        $data->where('start', '>=', $request->all());
+
+    }elseif($request->has('start.lte')){
+
+        $data->where('start', '<=', $request->all());
+
+    }elseif($request->has('end.eq')){
+
+        $data->where('end', '=', $request->all());
+
+    }elseif($request->has('end.gte')){
+
+        $data->where('end', '>=', $request->all());
+
+    }elseif($request->has('end.lte')){
+
+        $data->where('end', '<=', $request->all());
+
+    }elseif($request->has('limit') && $request->has('offset')){
+
+        $data->offset($request->limit)->limit($request->offset)->get();
```

Comment:
Change $data->offset($request->limit)->limit($request->offset)->get(); to $data->offset($request->offset)->limit($request->limit)->get();